open_spiel/algorithms/cfr.h

```diff
@@ -138,8 +166,15 @@ class CFRSolverBase {
     return std::unique_ptr<Policy>(new CFRCurrentPolicy(info_states_, nullptr));
   }
 
+  CFRInfoStateValuesTable& InfoStateValuesTable() { return info_states_; }
+
+  // See comments above CFRInfoStateValues::Serialize(double_precision) for
+  // notes about the double_precision parameter.
+  std::string Serialize(int double_precision = -1,
+                        std::string delimiter = "<~>") const;
+
  protected:
-  const Game& game_;
+  std::shared_ptr<const Game> game_;
```

Comment:
Can you explain why you need this to change to shared_ptr ?

Reply:
Because a game is deserialized by calling `LoadGame` within some deserialization function. I don't think it's possible to create a new solver instance and assigning a reference to the game without managing the game instance by the solver creator? Maybe I'm missing something here but if the deserialization function creates a smart pointer to the game instance and then sets the reference the game instance would get destroyed after the deserialization function exists and if the function creates a regular pointer, we have a memory leak. Is there a better approach?